### DISCUSSION TITLE
Achieving File Upload Synchronization from MinIO S3 to CKAN

### DISCUSSION THREAD

**Discussion by nir0sha** [ckan/ckan#7962] [Q&A]
I've successfully set up CKAN integrated with MinIO S3 for file storage. Presently, when files are uploaded within CKAN, they reflect in the associated MinIO S3 bucket. However, I'm curious about the feasibility of the reverse process: is it possible for files uploaded directly into the MinIO S3 bucket to automatically reflect or sync within the CKAN application?
I'm exploring ways to establish this bidirectional synchronization, where files uploaded to the S3 bucket trigger updates or reflections within the CKAN platform. Are there recommended methods, configurations, or tools available to achieve this file synchronization from MinIO S3 to CKAN seamlessly?

**Comment by pdelboca** [comment]
Hi! Interesting topic! I have no answer but I'm curious about it.

Can you elaborate a little bit more what do you mean with `synchronization` and describe an scenario?

How has been implemented the integration with MinIO? I can imagine that you save in the resource a URL or a pointer to the resource? If that's the case I imagine "something" will need to update the resource in CKAN whenever the bucket is updated. If the resource do not exist in CKAN, you will need to create it.

Are you expecting changes in a file already asociated with a CKAN resource to be reflected in the site? Or new Files in MinIO to appear in CKAN?
